LKSM Folder Archives Test Automation

labkey-danield opened the pull request:
#### Rationale
Updating various shared test code for folder archive testing in LKSM.

The FolderArchiveDialog extends the DeleteConfirmationDialog. It's not identical but similar in behavior. I thought is would easier and quicker to derive from the delete dialog than to create a "new" dialog. Especially since I doubt the archive dialog will be used outside of these tests.

#### Related Pull Requests
- https://github.com/LabKey/limsModules/pull/1490

#### Changes
- Added BaseWebDriverTest.TRICKY_DOMAIN_NAME_CHARACTERS (Maybe not needed anymore?)
- Add overloaded DeleteConfirmationDialog.confirmDelete to allow for different delete button text.
- Add FolderArchiveDialog
- Made public ResponsiveGrid.clickColumnMenuItem
- Correct locator in NavBar.

labkey-danield committed:
Add TRICKY_DOMAIN_NAME_CHARACTERS to BaseWebDriverTest.
Change pasteSamplesInput in BulkAddReferenceModalDialog from a method to use refind.
Update ContainerList to account for archived containers (folders).
Change ContainerList.getContainers to getAllContainers. Update usages.
Add DeleteConfirmationDialog.clickConfirmButton so class can be used by FolderArchiveDialog.
Add FolderArchiveDialog.
Change finders in FolderManagementPage to refind.
Add archive functionality to FolderSettingsPanel.
Change locator in NavBar. Now it can be used with Admin pages.
Add simple archive test.

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merging develop into fb_smFolderArchives.

labkey-danield committed:
Fix some bad merge resolutions.

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Added tests for searching, filter and file import.
Made public the ResponsiveGrid.clickColumnMenuItem

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-danield committed:
Remove unneeded constant.
Reverting changes from container to folder.

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives

labkey-tchad committed:
Merge remote-tracking branch 'origin/develop' into fb_smFolderArchives

labkey-tchad commented on src/org/labkey/test/components/ui/search/SampleFinder.java:
```diff
@@ -70,11 +70,13 @@ public void waitForReady()
     public EntityFieldFilterModal clickAddParent(String filterKind)
     {
+        // Why is this method called clickAddParent?
+
```
This name is a relic of when the functionality was limited to searching by "Registry Parent" or "Sample Parent". Perhaps a questionable method name at the time but has gotten progressively worse with time.

labkey-danield replied:
I've changed this to `clickAddSearchCard`. I think that name aligns it with more with the `removeSearchCard` and `removeAllSearchCards` methods. Let me know if yo disagree.

labkey-tchad commented on src/org/labkey/test/components/ui/FolderArchiveDialog.java:
```diff
@@ -0,0 +1,39 @@
+    public ConfirmPage clickYesArchive(Integer waitSeconds)
+    {
+        return super.clickConfirmButton(waitSeconds, "Yes, Archive Folder");
+    }
```
Extending `DeleteConfirmationDialog` is going to make this class have half a dozen "delete" related methods.
Possibly out of scope for this but we should pull a bunch of functionality from `DeleteConfirmationDialog` up into a more generic `ConfirmationDialog` so that all these confirmation dialogs aren't stepping on each other.

labkey-danield replied:
Yeah, I was being a little lazy with this. I've changed the `FolderArchiveDialog` to only extend the `ModalDialog`, and have reverted the changes to the `DeleteConfirmationDialog`.
If refactoring the `DeleteConfirmationDialog` is an important task we should discuss in stand up and see where it fits into the priority of other tasks.

labkey-susanh reviewed (commented):
Seems good to me. I agree with Trey's comment about refactoring to have a more generic `ConfirmationDialog`. I remember seeing the `DeleteConfirmationDialog` as a base for something else giving me pause during some other PR.

labkey-danield commented:
> Seems good to me. I agree with Trey's comment about refactoring to have a more generic `ConfirmationDialog`. I remember seeing the `DeleteConfirmationDialog` as a base for something else giving me pause during some other PR.

I've changed the FolderArchiveDialog to extend the ModalDialog and not the DeleteConfirmationDialog. I'll bring up in stand up the possibility of refactoring parts of the DeleteConfirmationDialog.

labkey-danield committed:
Change FolderArchiveDialog to extend ModalDialog and not DeleteConfirmationDialog.
Revert changes to DeleteConfirmationDialog.
In SampleFinder, change clickAddParent to clickAddSearchCard
In FolderManagementPage moved the finders out of the elementCache and into the appropriate methods.
Updated constructor of FolderSettingsPanel to take a FolderManagementPage and not a WebDriver
Consolidated some test code and added screenshots on failures.

labkey-danield committed:
Merge branch 'develop' into fb_smFolderArchives
